workspaces/ui-v2/src/store/diff/slice.ts

```diff
@@ -0,0 +1,56 @@
+import { createSlice, SerializedError } from '@reduxjs/toolkit';
+
+import { IValueAffordanceSerializationWithCounterGroupedByDiffHash } from '@useoptic/cli-shared/build/diffs/initial-types';
+import { IUnrecognizedUrl } from '@useoptic/spectacle';
+
+import { ParsedDiff } from '<src>/lib/parse-diff';
+import { AsyncStatus } from '<src>/types';
+import { fetchDiffsForCapture } from './thunks';
+
+// This needs to be exported for typescript to be able to infer typings
+export type DiffState = {
+  state: AsyncStatus<
+    {
+      diffs: ParsedDiff[];
+      urls: IUnrecognizedUrl[];
+      trails: IValueAffordanceSerializationWithCounterGroupedByDiffHash;
```

Comment:
Bonus points are available for renaming this horrible type. Or using an anonymous type that composes from it's components. Anything 😂😭

Reply:
Ok I attempted to rename these - I think `IAffordanceTrails` and `IAffordanceTrailsDiffHashMap` are slightly better - but I'm not sure what these are _actually_ representing so there's probably a better name